cms/release_calendar/forms.py

```diff
@@ -92,3 +98,27 @@ def validate_english_release_date_text_format(self, text: str) -> None:
                     )
                 }
             ) from None
+
+    def validate_bundle_not_pending_publication(self, status: str) -> None:
+        if self.instance.status == status:
+            return
+
+        bundle = self.instance.active_bundle
+        if not (bundle and bundle.is_ready_to_be_published):
+            return
+
+        if self.for_user and user_can_manage_bundles(self.for_user):
+            bundle_str = format_html(
+                '<a href="{}" target="_blank" title="Manage bundle">{}</a>',
+                reverse("bundle:edit", args=[bundle.pk]),
+                bundle.name,
+            )
+        else:
+            bundle_str = bundle.name
+
+        message = format_html(
+            "This release calendar page is linked to bundle '{}' which is ready to be published. "
+            "Please unschedule the bundle and unlink the release calendar page before making the cancellation.",
+            bundle_str,
+        )
```

Comment:
I suggest this is done for cancellations and also date changes. The date change implication is currently not visible to the users right now.

Reply:
Let's cover this in CMS-428